### PR TITLE
Added Euclidean GCD [JavaScript]

### LINKED ISSUE #297: Euclidean GCD [JavaScript]
#### This issue is about [Euclidean GCD](https://en.wikipedia.org/wiki/Euclidean_algorithm).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.

### PR DESCRIPTION
**Pull request by Anshuman-Verma**
Fixes #297 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: euclidean_gcd/euclidean_gcd.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+function euclidean_gcd_recursive(first, second) {
```
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.js]: Please follow standard naming conventions.

File: euclidean_gcd/euclidean_gcd.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	if(second === 0) {
+		return first;
+	} else {
+		return euclidean_gcd_recursive(second, (first % second));
+	}
+}
+
+function euclidean_gcd(first, second) {
```
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.js]: Same here, follow standard naming conventions.

File: euclidean_gcd/euclidean_gcd.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	while(second != 0) {
```
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.js]: Isn't this written as  `!==`  in `js` ?

File: euclidean_gcd/euclidean_gcd.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+function main() {
+	let first = 20;
+	let second = 30;
+	console.log('Recursive GCD for %d and %d is %d', first, second ,euclidean_gcd_recursive(first, second));
+	console.log('Iterative GCD for %d and %d is %d', first, second, euclidean_gcd(first, second));
+}
+
+main();
```
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.js]: Add a newline at end.

File: euclidean_gcd/euclidean_gcd.js (hunk 1 above)
Comment (Anshuman-Verma) [inline euclidean_gcd/euclidean_gcd.js]: Please explain the standard naming convention, I've just followed what I got from the C file of this algo ? That will mean C file will require change too ?

File: euclidean_gcd/euclidean_gcd.js (hunk 3 above)
Comment (Anshuman-Verma) [inline euclidean_gcd/euclidean_gcd.js]: Both of them will work, you can compile and try.

File: euclidean_gcd/euclidean_gcd.js (hunk 4 above)
Comment (Anshuman-Verma) [inline euclidean_gcd/euclidean_gcd.js]: FIXED

File: euclidean_gcd/euclidean_gcd.js (hunk 3 above)
Comment (Anshuman-Verma) [inline euclidean_gcd/euclidean_gcd.js]: Though !== is strict comparison and should be followed here, hence updated :+1:

File: euclidean_gcd/euclidean_gcd.js (hunk 1 above)
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.js]: No, the naming convention in java and javascript is `camelCase`, while c, cpp and python has underscores.

**Comment by Monal5031** [comment]
@Anshuman-Verma Codacy bot is also pointing the same thing, function and variable name should be `camelCase`. Please fix that also.

**Comment by Anshuman-Verma** [comment]
Alright, I'll do it @Monal5031 
But we would need to update function names in c and java files as well since they used this function nomenclature which i used. 
Take a look into it too plz :+1:

**Comment by Monal5031** [comment]
@Anshuman-Verma Please check again java is following `camelCase`. And I already told you c cpp and python files follow underscores not `camelCase` .

File: euclidean_gcd/euclidean_gcd.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+function euclideanGcdRecursive(first, second) {
+	if(second === 0) {
+		return first;
+	} else {
+		return euclideanGcdRecursive(second, (first % second));
+	}
+}
+
+function euclideanGcd(first, second) {
```
Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.js]: This should be `euclideanGcdIterative (...) {`

**Comment by Monal5031** [comment]
@Anshuman-Verma Fix travis issues and also add comments about params and return values please.

**Comment by Anshuman-Verma** [comment]
Thanks for review @Monal5031 
I  guess I've covered everything. :+1:

**Comment by aashutoshrathi** [comment]
Update Branch

File: euclidean_gcd/euclidean_gcd.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+function euclideanGcdRecursive (first, second) {
```
Comment (aviaryan) [inline euclidean_gcd/euclidean_gcd.js]: Function name should have GCD instead of Gcd

File: README.md [hunk 7]
```diff
@@ -21,7 +21,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Counting Sort](http://www.geeksforgeeks.org/counting-sort/)| [:white_check_mark:](counting_sort/counting_sort.c) | | [:white_check_mark:](counting_sort/CountingSort.java) | | | |
 | [Depth First Traversal](http://www.geeksforgeeks.org/depth-first-traversal-for-a-graph/) | | | [:white_check_mark:](depth_first_traversal/DepthFirstTraversal.java) | | | |
 | [Dijkstra Algorithm](https://en.wikipedia.org/wiki/Dijkstra's_algorithm/) | [:white_check_mark:](dijkstra/dijkstra.c) | | [:white_check_mark:](dijkstra/Dijkstra.java) | [:white_check_mark:](dijkstra/dijkstra.py) | | |
-| [Euclidean GCD](https://en.wikipedia.org/wiki/Euclidean_algorithm) | [:white_check_mark:](euclidean_gcd/euclidean_gcd.c) | | [:white_check_mark:](euclidean_gcd/EuclideanGCD.java) | [:white_check_mark:](euclidean_gcd/euclidean_gcd.py) | | |
+| [Euclidean GCD](https://en.wikipedia.org/wiki/Euclidean_algorithm) | [:white_check_mark:](euclidean_gcd/euclidean_gcd.c) | | [:white_check_mark:](euclidean_gcd/EuclideanGCD.java) | [:white_check_mark:](euclidean_gcd/euclidean_gcd.py) | | [:white_check_mark:](euclidean_gcd/euclidean_gcd.js) |
```
Comment (aashutoshrathi) [inline README.md]: Change file name to euclideanGCD.js

Comment (aviaryan) [inline README.md]: Or rather euclideanGCD.js

File: euclidean_gcd/euclidean_gcd.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+	/*
```
Comment (aashutoshrathi) [inline euclidean_gcd/euclidean_gcd.js]: Add Time Complexity

File: euclidean_gcd/euclidean_gcd.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+	*/
+	if (second === 0) {
+		return first;
+	} else {
+		return euclideanGcdRecursive(second, (first % second));
+	}
+}
+
+function euclideanGcdIterative (first, second) {
+	/*
```
Comment (aashutoshrathi) [inline euclidean_gcd/euclidean_gcd.js]: Add Time Complexity

File: README.md (hunk 7 above)
Comment (Anshuman-Verma) [inline README.md]: FIXED

Comment (aashutoshrathi) [inline README.md]: Nup Not fixed in README.md still

File: euclidean_gcd/euclidean_gcd.js (hunk 8 above)
Comment (Anshuman-Verma) [inline euclidean_gcd/euclidean_gcd.js]: Time complexity is mentioned in neither of the implementations.

File: README.md (hunk 7 above)
Comment (Anshuman-Verma) [inline README.md]: Yea updated it now

**Pull request merged**